components/dashboard/src/CookieSettingsModal.tsx

```diff
@@ -0,0 +1,108 @@
+            analytical: cookiePreferences.analytical,
+            targeting: cookiePreferences.targeting,
+        });
+
+        props.onSave();
+    };
+
+    const onClose = () => {
+        const value = getConsentCookies();
+        if (value) {
+            setCookiePreferences(JSON.parse(value));
+        }
+
+        props.onClose();
+    };
+
+    return (
+        <Modal visible={props.showCookieSettings} onClose={onClose}>
+            <h3 className="mb-4">Cookie settings</h3>
+            <div className="flex gap-2">
+                <h4>Strictly necessary cookies</h4>
+                <div className="-mt-4">
+                    <CheckBox title="" desc={undefined} checked={true} disabled={true} name="necessary" />
+                </div>
+            </div>
+            <p className="mb-4">
+                These are cookies that are required for the operation of our Site and under our terms with you. They
+                include, for example, cookies that enable you to log into secure areas of our Site or (on other sites)
+                use a shopping cart or make use of e-billing services.
+            </p>
+            <div className="flex gap-2">
+                <h4>Analytical / Performance cookies</h4>
+                <div className="-mt-4">
+                    <CheckBox
+                        title=""
+                        desc={undefined}
+                        checked={cookiePreferences.analytical}
+                        name="analytical"
+                        onChange={handleAnalyticalCheckboxChange}
+                    />
+                </div>
+            </div>
+            <p className="mb-4">
+                These allow us to recognise and count the number of visitors and to see how visitors move around our
+                site when they are using it. This helps us improve the way our Website works, for example, by ensuring
+                that users are finding what they are looking for easily.
+            </p>
+            <div className="flex gap-2">
+                <h4>Targeting/Advertising cookies</h4>
+                <div className="-mt-4">
+                    <CheckBox
+                        title=""
+                        desc={undefined}
+                        checked={cookiePreferences.targeting}
+                        name="targeting"
+                        onChange={handleTargetingCheckboxChange}
+                    />
+                </div>
+            </div>
+            <p className="mb-4">
+                These cookies record your visit to our Website, the pages you have visited and the links you have
+                followed. We will use this information subject to your choices and preferences to make our Website and
+                the advertising displayed on it more relevant to your interests. We may also share this information with
+                third parties for this purpose.
+            </p>
+            <button onClick={onSave}>Save</button>
```

Comment:
**issue**: In most, if not, all other modals, we're grouping the primary and secondary actions towards the right side of the modal. Aligning the primary action here on the left could affect memory muscle.

Reply:
Please don't let me hijack the discussion 🙏 

I just wanted to point out that this comment and [the one about checkboxes](https://github.com/gitpod-io/gitpod/pull/9299#discussion_r852992789) immediately make the words ["encapsulation" and "design system"](https://discord.com/channels/816244985187008514/914836798830690344/915363278031577109) light up in my mind like [big fire signals on the sands of a remote island at night](https://bluejayblog.files.wordpress.com/2014/06/sos-aregua.jpg) 😄

Reply:
Thanks for chiming in, @trumbitta! ✨ 

Here are some benefits of the current status quo:
1. Running into such cases through code reviews helps improve our team by exposing team members to design thinking and why some options are better than others.
2. Keeping existing components loose can breed new component structures that are needed as the product matures. Using this approach means embracing technical debt and continuously iterating. For example, the alert component improved over time because of these loose ends, and eventually we managed to introduce a better and more flexible component instead of maintaining two separate components, see https://github.com/gitpod-io/gitpod/issues/9411.
3. Focusing on shipping this as a [minimum viable change (MVC)](https://www.notion.so/gitpod/Core-Values-Differentiators-2ed4c2f93c84499b98e3b5389980992e#0c2e0d6c8a99404b97ccdc206b573d72), even if the button was never aligned to the right before merging this, sounds optimal as we're always making the quickest change possible to improve the user's outcome.